<?xml version="1.0"?>

<gazebo:world xmlns:xi="http://www.w3.org/2001/XInclude"
	xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz"
	xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model"
	xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
	xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
	xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom"
	xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint"
	xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
	xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering"
	xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable"
	xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
	xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics">
	
	<verbosity>5</verbosity>
	
	<physics:ode>
		<stepTime>0.03</stepTime>
		<gravity>0 0 -9.8</gravity>
		<cfm>10e-5</cfm>
		<erp>0.8</erp>
	</physics:ode>
	
	<rendering:gui>
		<type>fltk</type>
		<size>800 600</size>
		<pos>0 0</pos>
	</rendering:gui>
	
	<rendering:ogre>
		<ambient>1.0 1.0 1.0 1.0</ambient>
		<sky>
			<material>Gazebo/CloudySky</material>
		</sky>
		<grid>false</grid>
	</rendering:ogre>
	
	<model:physical name="plane1_model">
		<xyz>0 0 0</xyz>
		<rpy>0 0 0</rpy>
		<static>true</static>
		<body:plane name="plane1_body">
			<geom:plane name="plane1_geom">
				<normal>0 0 1</normal>
				<size>3 2</size>
				<segments>10 10</segments>
				<uvTile>100 100</uvTile>
				<material>Gazebo/LightWood</material>
			</geom:plane>
		</body:plane>
	</model:physical>
	
	<model:physical name="Observercam_model">
		<xyz>0 0 2.65</xyz>
		<rpy>90 90 0</rpy>
		<static>true</static>
		<body:empty name="Observercam_body">
			<sensor:camera name="Observercam_sensor">
				<imageSize>800 600</imageSize>
				<hfov>60</hfov>
				<nearClip>0.1</nearClip>
				<farClip>100</farClip>
				<saveFrames>false</saveFrames>
				<saveFramePath>frames</saveFramePath>
			</sensor:camera>
		</body:empty>
	</model:physical>
	
	<model:physical name="wall_model">
		<xyz>0 0 0</xyz>
		<static>true</static>
		<body:empty name="wall_body">
			<geom:box name="wall_geom1">
				<xyz>1.5 0 0</xyz>
				<size>0.01 2 0.1</size>
				<visual>
					<scale>0.01 2 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			<geom:box name="wall_geom2">
				<xyz>-1.5 0 0</xyz>
				<size>0.01 2 0.1</size>
				<visual>
					<scale>0.01 2 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			<geom:box name="wall_geom3">
				<xyz>0 1 0</xyz>
				<size>3 0.01 0.1</size>
				<visual>
					<scale>3 0.01 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			<geom:box name="wall_geom4">
				<xyz>0 -1 0</xyz>
				<size>3 0.01 0.1</size>
				<visual>
					<scale>3 0.01 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			</body:empty>
		    <body:empty name="corner1">
				<rpy>0 0 45</rpy>
				<xyz>1.4 -0.9 0</xyz>
				<geom:box name="wall_corner1">
				
				<size>0.29 0.01 0.1</size>
				<visual>
					<scale>0.29 0.01 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			</body:empty>
		<body:empty name="corner2">
				<rpy>0 0 -45</rpy>
				<xyz>-1.4 -0.9 0</xyz>
				<geom:box name="wall_corner2">
				
				<size>0.29 0.01 0.1</size>
				<visual>
					<scale>0.29 0.01 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			</body:empty>
		<body:empty name="corner3">
				<rpy>0 0 -45</rpy>
				<xyz>1.4 0.9 0</xyz>
				<geom:box name="wall_corner3">
				
				<size>0.29 0.01 0.1</size>
				<visual>
					<scale>0.29 0.01 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			</body:empty>
		<body:empty name="corner4">
				<rpy>0 0 45</rpy>
				<xyz>-1.4 0.9 0</xyz>
				<geom:box name="wall_corner4">
				
				<size>0.29 0.01 0.1</size>
				<visual>
					<scale>0.29 0.01 0.1</scale>
					<mesh>unit_box</mesh>
					<material>Gazebo/BumpyMetal</material>
				</visual>
			</geom:box>
			</body:empty>
		
	</model:physical>
	
	<model:physical name="robot1">
		<xyz>-0.1 0.2 0.0</xyz>
		<rpy>0 0 -10</rpy>
		<static>false</static>

		<model:physical name="camera">	
  		        <include embedded="true">
			        <xi:include href="models/epuck_camera.model"/>
		        </include>
                </model:physical>
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot2">
		<xyz>0.1 0.2 0.0</xyz>
		<rpy>0 0 -36</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot3">
		<xyz>-0.3 0.2 0.0</xyz>
		<rpy>0 0 180</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot4">
		<xyz>0.3 0.2 0.0</xyz>
		<rpy>0 0 -90</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot5">
		<xyz>-0.5 0.2 0.0</xyz>
		<rpy>0 0 90</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot6">
		<xyz>0.5 0.2 0.0</xyz>
		<rpy>0 0 0</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot7">
		<xyz>-0.7 0.2 0.0</xyz>
		<rpy>0 0 -220</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot8">
		<xyz>0.7 0.2 0.0</xyz>
		<rpy>0 0 220</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot9">
		<xyz>-0.1 -0.2 0.0</xyz>
		<rpy>0 0 -120</rpy>
		<static>false</static>
	
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot10">
		<xyz>0.1 -0.2 0.0</xyz>
		<rpy>0 0 120</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot11">
		<xyz>-0.3 -0.2 0.0</xyz>
		<rpy>0 0 -50</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot12">
		<xyz>0.3 -0.2 0.0</xyz>
		<rpy>0 0 50</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot13">
		<xyz>-0.5 -0.2 0.0</xyz>
		<rpy>0 0 -80</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot14">
		<xyz>0.5 -0.2 0.0</xyz>
		<rpy>0 0 80</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot15">
		<xyz>-0.7 -0.2 0.0</xyz>
		<rpy>0 0 20</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
	<model:physical name="robot16">
		<xyz>0.7 -0.2 0.0</xyz>
		<rpy>0 0 -20</rpy>
		<static>false</static>
		
		<include embedded="true">
			<xi:include href="models/epuck_new.model"/>
		</include>
	</model:physical>
	
</gazebo:world>
